cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -812,4 +815,60 @@ public static List<String> generateAllServiceDnsNames(String namespace, String s
 
         return dnsNames;
     }
+
+    /**
+     * Validate CPU requests and limits.
+     * 
+     * @param resources Compute resources configuration.
+     * @param owner Compute resources owner.
+     * 
+     * @return Error set.
+     */
+    public static Set<String> validateCpuResources(ResourceRequirements resources, String owner) {
+        return validateComputeResources(resources, "cpu", owner);
+    }
+
+    /**
+     * Validate memory requests and limits.
+     * 
+     * @param resources Compute resources configuration.
+     * @param owner Compute resources owner.
+     * 
+     * @return Error set.
+     */
+    public static Set<String> validateMemoryResources(ResourceRequirements resources, String owner) {
+        return validateComputeResources(resources, "memory", owner);
+    }
+
+    /**
+     * Validate compute resources requests and limits.
+     * 
+     * TODO: handle any type, not just cpu and memory.
+     * 
+     * @param resources Compute resources configuration.
+     * @param type Compute resources type.
+     * @param owner Compute resources owner.
+     * 
+     * @return Error set.
+     */
+    private static Set<String> validateComputeResources(ResourceRequirements resources, String type, String owner) {
+        Set<String> errors = new HashSet<>();
+        String ownerId = owner != null && !owner.isBlank() ? owner : "unknown";
+        if (resources != null && resources.getRequests() != null) {
+            Quantity cpuRequest = resources.getRequests().get(type);
```

Comment:
Its a generic method, so maybe a generic name instead of `cpuRequest` would be better?